cpp/src/arrow/compute/type_fwd.h

```diff
@@ -50,14 +50,6 @@ struct KernelState;
 
 struct Declaration;
```

Comment:
I think `Declaration` is part of `exec` too.

Reply:
good point, moving in `exec/type_fwd.h`